lib/drivers/mock/mock.go

```diff
@@ -1,26 +1,37 @@
 package mock
 
 import (
+	"errors"
+
 	"github.com/PacketFire/immigrant/lib/core"
 )
 
 type MockDriver struct {
 	Revisions []core.Revision
-	State     core.Revision
 }
 
 func (this *MockDriver) Init(config map[string]string) error {
 	return nil
 }
 
-func Migrate(r core.Revision, ec chan error) {
+func (this *MockDriver) Migrate(r *core.Revision, ec chan error) {
+	this.Revisions = append(this.Revisions, *r)
+	ec <- nil
 }
-
-func Rollback(r core.Revision, ec chan error) {
+func (this *MockDriver) Rollback(r *core.Revision, ec chan error) {
+	if len(this.Revisions) <= 0 {
```

Comment:
the length of an array is always guaranteed to be non-negative. this check is valid but it includes states that can never exist